File: src/test/java/kimspring/splearn/domain/member/ProfileTest.java
package kimspring.splearn.domain.member;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

class ProfileTest {
    @Test
    void profile() {
        new Profile("kim");
        new Profile("kim2");
        new Profile("0044");
        new Profile("");
    }

    @Test
    void profileFail() {
        assertThatThrownBy(() -> new Profile("toolongtoolongtoolong"))
                .isInstanceOf(IllegalArgumentException.class);
        assertThatThrownBy(() -> new Profile("KIM"))
                .isInstanceOf(IllegalArgumentException.class);
        assertThatThrownBy(() -> new Profile("킴"))
                .isInstanceOf(IllegalArgumentException.class);
    }
    
    @Test
    void url() {
        Profile profile = new Profile("kim");

        assertThat(profile.url()).isEqualTo("@kim");
    }

}

File: src/test/kotlin/kimspring/splearn/domain/member/ProfileTest.kt
package kimspring.splearn.domain.member

import io.kotest.assertions.throwables.shouldThrow
import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe

class ProfileTest :
    FunSpec({
        test("profile") {
            Profile("kim")
            Profile("kim2")
            Profile("0044")
            Profile("")
        }

        test("profileFail") {
            shouldThrow<IllegalArgumentException> { Profile("toolongtoolongtoolong") }
            shouldThrow<IllegalArgumentException> { Profile("KIM") }
            shouldThrow<IllegalArgumentException> { Profile("킴") }
        }

        test("url") {
            val profile = Profile("kim")
            profile.url() shouldBe "@kim"
        }
    })
